src/app/routes/enrolment/enrolment-list/enrolment-list.component.ts

```diff
@@ -247,12 +247,18 @@ export class EnrolmentListComponent implements OnInit, OnDestroy {
         this.getList(this.dynamicRejected, this.dynamicAccepted);
       });
   }
-
-  addToDidDoc(element: EnrolmentClaim) {
+  async addToDidDoc(element: EnrolmentClaim) {
```

Comment:
This logic should be done in `PublishRoleService`, not in enrolment-list component, because we have 3 places from where we can publish role. This solves only one place, but all of them are using publish-role.service.ts file.
All places from where you can publish role:
1. Enrol page
2. Creating a VC
3. My enrolments list,

and all of them are using `PublishRoleService`.

Reply:
Thanks @dawidgil see below where I moved to publish role service.